tracing-opentelemetry/src/main/java/org/axonframework/tracing/opentelemetry/OpenTelemetrySpan.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing.opentelemetry;
+
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Scope;
+import org.axonframework.tracing.Span;
+
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+/**
+ * {@link Span} implementation that uses OpenTelemetry's {@link io.opentelemetry.api.trace.Span} to provide tracing
+ * capabilities to an application.
+ * <p>
+ * These traces should always be created using the {@link OpenTelemetrySpanFactory} since this will make sure the proper
+ * parent context is extracted before creating the {@link Span}.
+ */
+public class OpenTelemetrySpan implements Span {
+
+    private final SpanBuilder spanBuilder;
+    private io.opentelemetry.api.trace.Span span;
+    private Scope activeScope;
+
+    /**
+     * Creates the span, based on the {@link SpanBuilder} provided. This {@link SpanBuilder} will supply the
+     * {@link io.opentelemetry.api.trace.Span} when the {@link #start()} method is invoked.
+     *
+     * @param spanBuilder The provider of the {@link io.opentelemetry.api.trace.Span}.
+     */
+    OpenTelemetrySpan(SpanBuilder spanBuilder) {
+        this.spanBuilder = spanBuilder;
```

Comment:
Why not validate whether the given `spanBuilder` is not-null through the `BuilderUtils`?

Reply:
The `BuilderUtils` throw a `ConfigurationException`, which I think does not apply here. Additionally, it's package-private, so it should not be accessed by others

Reply:
I'd rather have another exception than the chance a `NullPointerException` is thrown because, one way or another, the `SpanBuilder` is `null`. 
The `BuilderUtils` operations is one way to achieve that. If you feel strongly that construction isn't part of configuration, then another exception is fine by me too.

Reply:
Updated it to check for nullness explicitly.